src/oss/langchain-models.mdx

```diff
@@ -2,14 +2,605 @@
+
+TODO a switcher to iterate through all different tool schema options
+TODO model switcher to show this example for our top 5 providers
+
+```python Binding tools highlight={16}
+from pydantic import BaseModel, Field
+from langchain.chat_models import init_chat_model
+
+class GetWeather(BaseModel):
+    """Get the current weather in a given location"""
+    location: str = Field(..., description="The city and state, e.g. Boston, MA")
+
+class Calculator(BaseModel):
+    """Perform mathematical calculations"""
+    expression: str = Field(..., description="Mathematical expression to evaluate")
+
+model = init_chat_model(
+    model="gpt-5", 
+    model_provider="openai"
+)
+model_with_tools = model.bind_tools([GetWeather, Calculator])
+
+response = model_with_tools.invoke("What's the weather like in Boston and what's 25 * 4?")
+for tool_call in response.tool_calls:
+    # View tool calls made by the model
+    print(f"Tool: {tool_call['name']}")
+    print(f"Args: {tool_call['args']}")
+```
+
+In this example, the model's response includes a **request** to execute a tool.
+It is up to you to perform the requested action and return the result back to
+the model for use in subsequent reasoning.
+
+TODO link out to a different guide on tool execution; beyond the scope here
+
+#### Forcing tool calls
+
+By default, the model has the freedom to choose which bound tool to use based on
+the user's input. However, you might want to force choosing a tool, ensuring the
+model uses either a particular tool or *any* tool from a given list:
+
+<CodeGroup>
+
+```python Force use of specific tools
+response = model_with_tools.invoke(
+    "What's the weather in both New York and London, and calculate 15 + 25",
+    force_tool_calls=["GetWeather", "Calculator"]
+)
+```
+
+```python Force use of any tool
+response = model_with_tools.invoke(
+    "What's the weather in both New York and London, and calculate 15 + 25",
+    force_tool_calls="any"
+)
+```
+
+</CodeGroup>
+
+
+#### Parallel tool calls
+
+TODO */}
+
+### Structured outputs
+
+Models can be requested to provide their response in a format matching a given
+schema. This is useful for ensuring the output can be easily parsed and used in
+subsequent processing.
+
+```python Structured output highlight={12-14}
+from pydantic import BaseModel, Field
+
+class Movie(BaseModel):
+    title: str = Field(..., description="The title of the movie")
+    year: int = Field(..., description="The year the movie was released")
+    director: str = Field(..., description="The director of the movie")
+    rating: float = Field(..., description="The movie's rating out of 10")
+
+model_with_structure = model.with_structured_output(Movie)
+response = model_with_structure.invoke("Provide details about the movie Inception")
+print(response)  # Movie(title="Inception", year=2010, director="Christopher Nolan", rating=8.8)
+```
+
+See the [structured outputs]() guide for more info.
```

Comment:
structured outputs is about structured outputs in agents, not models. any relevant info needs to go here